pkg/queue/concurrency_state.go

```diff
@@ -19,30 +19,45 @@ package queue
 import (
 	"bytes"
 	"fmt"
+	"io/ioutil"
 	"net/http"
 	"os"
 	"sync"
+	"time"
 
 	"go.uber.org/atomic"
 	"go.uber.org/zap"
 )
 
 //nolint:gosec // Filepath, not hardcoded credentials
 const ConcurrencyStateTokenVolumeMountPath = "/var/run/secrets/tokens"
+const ConcurrencyStateTokenName = "state-token"
+const ConcurrencyStateToken = ConcurrencyStateTokenVolumeMountPath + "/" + ConcurrencyStateTokenName
 
 // ConcurrencyStateHandler tracks the in flight requests for the pod. When the requests
 // drop to zero, it runs the `pause` function, and when requests scale up from zero, it
 // runs the `resume` function. If either of `pause` or `resume` are not passed, it runs
 // the respective local function(s). The local functions are the expected behavior; the
 // function parameters are enabled primarily for testing purposes.
-func ConcurrencyStateHandler(logger *zap.SugaredLogger, h http.Handler, pause, resume func() error) http.HandlerFunc {
+func ConcurrencyStateHandler(logger *zap.SugaredLogger, h http.Handler, pause, resume func(*Token) error, tokenMountPath string) http.HandlerFunc {
 
 	var (
 		inFlight = atomic.NewInt64(0)
 		paused   = true
 		mux      sync.RWMutex
 	)
 
+	// keep serviceAccountToken up-to-date
+	var tokenCfg Token
+	refreshToken(&tokenCfg, tokenMountPath)
+	go func() {
+		for {
+			time.Sleep(1 * time.Minute)
```

Comment:
Rather than doing the token stuff here, what do you think about pulling it out in to the Pause and Resume request structs? They're already abstracting how the Handler should contact the endpoint, so it seems odd (to me) for Handler to have the token code rather than them (I'm imagining something like a `type concurrencyStateRequest struct {  token atomic.Value }` struct with Pause and Resume functions).

Reply:
It did cross my mind, though I was  leaning towards having the Pause/Resume functions grab the refreshed token when called, rather than running a goroutine in the handler to keep it refreshed. I had a little nagging worry that getting the token in the pause/resume functions might end up grabbing the wrong value vs. getting it in the handler (though to be fair it was a fleeting thought and I didn't really pursue it too far). That said, let me give it another go.

(Figures that I juuuuust dropped the request struct from the [previous PR](https://github.com/knative/serving/pull/11802#discussion_r713030012) :smile: )

Reply:
@julz before I go too far down the refactor train, would something like this be along these lines of what you're thinking?

```golang
type ConcurrencyEndpoint struct {
	endpoint string
	token atomic.Value
}

func (c ConcurrencyEndpoint) Pause() error { c.Request("pause") }

func (c ConcurrencyEndpoint) Resume() error { c.Request("resume") }

func (c ConcurrencyEndpoint) Request(request string) func() error {
	return func() error { /* do stuff */ }
}
```

Reply:
Turns out the end of the line wasn't all that far away... got this added in the last batch of commits